src/main/java/org/delivery/KiwiEats/models/ProductDTO.java

```diff
@@ -0,0 +1,43 @@
+package org.delivery.KiwiEats.models;
+
+import jakarta.validation.constraints.*;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.delivery.KiwiEats.entities.Category;
+import org.hibernate.validator.constraints.Length;
+import org.hibernate.validator.constraints.URL;
+
+import java.sql.Timestamp;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+public class ProductDTO {
+    @NotNull
+    private Long id;
+
+    @NotNull
+    @NotBlank(message = "Product Name cannot be empty")
+    @Length(max=200)
+    private String productName;
+
+    @NotNull
+    @URL(protocol = "https",message = "Image is not in the right format!")
+    private String productImage;
+
+    @NotNull
```

Comment:
no null needed because the frontend will make sure a proper value is sent back to the backend